docs/USAGE-GUIDE.md

```diff
@@ -1401,6 +1401,23 @@ ssize_t s2n_client_hello_get_extension_by_id(struct s2n_client_hello *ch, s2n_tl
 **s2n_client_hello_get_extension_length** returns the number of bytes the given extension type takes on the ClientHello message received by the server; it can be used to allocate the **out** buffer.
 **s2n_client_hello_get_extension_by_id** copies into the **out** buffer **max_length** bytes of a given extension type on the ClientHello and returns the number of copied bytes.
 
+### s2n\_client\_hello\_get\_session\_id
+
+```c
+int s2n_client_hello_get_session_id_length(struct s2n_client_hello *ch, uint32_t *out_length);
+int s2n_client_hello_get_session_id(struct s2n_client_hello *ch, uint8_t *out, uint32_t *out_length, uint32_t max_length);
+```
+
+- **ch** The s2n_client_hello on the s2n_connection. The handle can be obtained using **s2n_connection_get_client_hello**.
+- **out** Pointer to a buffer into which the session id bytes should be copied.
+- **out_length** Pointer to a **uint32_t** which for **s2n_client_hello_get_session_id_length** will be used to store the full length of session_id and for **s2n_client_hello_get_session_id** will be used to store the number of bytes copied to the **out** buffer.
+- **max_length** Max number of bytes to copy into the **out** buffer.
```

Comment:
Not all of the other documentation entries discuss arguments separately from the methods. Since out_length has two different uses, is it worth trying to discuss both at once? This section might not be helpful

Reply:
Fair enough - followed previous docs too much. I'll split it up.

Reply:
Got rid of the list and moved into description of individual function.